### DISCUSSION TITLE
Whisper Unity3d bindings

### DISCUSSION THREAD

**Discussion by Macoron** [ggml-org/whisper.cpp#673] [Bindings]
First of all, thanks to all developers of whisper.cpp. You are doing great job by making this open source and fast!

I've just finished bindings for Unity3d and place them in this repository called [whisper.unity](https://github.com/Macoron/whisper.unity). It wraps native C++ methods of whisper.cpp into simplified C# API that can be used for creating 3d interactive software in Unity (games, 3d chatbots, VR apps, etc). Not all original whisper methods and parameters are properly supported yet, but I hope it will be fixed relatively soon.

Right now it supports Windows, MacOS, Linux, iOS and Android. Maybe WebGL in future releases.

**Comment by Macoron** [comment]
Few updates with a new features for Unity bindings.

https://user-images.githubusercontent.com/6161335/235007367-940988f6-3160-4b33-bbd2-802e185c451f.mp4

*`whisper.tiny` in subtitles demo, color shows confidence level for each token*

https://user-images.githubusercontent.com/6161335/231581911-446286fd-833e-40a2-94d0-df2911b22cad.mp4

*"whisper-small.bin" model tested in English, German and Russian from microphone*

Recently I added custom Github Actions to build whisper.cpp from sources and update binary libraries of Unity bindings. I think it might be useful for other bindings developers, so I will leave pipeline here in case someone want to reuse it.

https://github.com/Macoron/whisper.unity/blob/master/.github/workflows/build_cpp.yml

**Comment by akareactor** [comment]
Thanks for your efforts! Is there any demo project for Whisper?

**Reply by Macoron** [reply]
Not compiled, but you can just download repo and run it using Unity Editor. There are samples scenes for AudioClips transcription, microphone input, subtitles mode, etc.

https://github.com/Macoron/whisper.unity

**Reply by akareactor** [reply]
Thank you very much!
Link https://github.com/Macoron/whisper.unity.git?path=/Packages/com.whisper.unity has no samples in package, downloaded it from page
